packages/upswyng-server/EVENTLOG.md

```diff
@@ -0,0 +1,154 @@
+# Event Logs
+
+UpSwyng server has a framework used to record business events happening in the system
+like Resources being created or Alerts going live. Events are recorded in the database
+and are shown in the UI accessable by administrators. When an event is recorded the
+`upswyngbot` on Slack can also send a message to its group.
+
+**See [PR #326](https://github.com/CodeForBoulder/upswyng/pull/326) for the implementation of the steps in this document**
+
+## Creating a new Event Log kind
+
+For this example, we'll setup logging for when an Alert goes live. The worker job checks regularly
+for Alerts whose start time has recently passed and marks them as "processed" (eventually it will send
+push notifications).
+
+### Create types
+
+The types for Event Logs are located in `TEventLog.ts` in the `@upswyng/upswyng-types`
+package. The `TEventLog` interface represents any Event Log. It contains the user who initiated the action
+which caused the event, as well as the kind of the event (we'll use `alert_live` for this case), plus extra
+details which are specific to that kind of event.
+
+To start, add the kind name to the `EventLogKind` object in `TEventLog.ts`:
+
+```typescript
+export const EventLogKind = {
+  // .....
+  alert_live: null,
+  // .....
+};
+export type TEventLogKind = keyof typeof EventLogKind;
```

Comment:
You can loop through enum values as well, right? If so, is there a benefit to this approach over using an enum?

Reply:
errrr lemme look at this more.. I might be pulling this from flow and enums arent exactly the same

Reply:
u right u right.. trying to flow it